WIP: showing anon users in stats

eleijonmarck opened the pull request:
<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/grafana/blob/main/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the main branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

7. If your PR content should be added to the What's New document for the next major or minor release, add the **add to what's new** label to your PR. Note that you should add this label to the main PR that introduces the feature; do not add this label to smaller PRs for the feature.

-->

**What is this feature?**
- Adding anon stats to usagestats
- Consumption of anon usage for frontend to show the anon users

[Add a brief description of what the feature or update does.]

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

sd2k committed:
Alerting: Add actions extension point to alert instances table view (#77900)

This PR adds a new [extension point][] to each row of the
alert instances table. This allows plugins to add actions
to a dropdown menu in the rightmost column of the table.
Those actions are passed the alert instance so can use it
for contextual handling.

See https://github.com/grafana/machine-learning/pull/3461
for an example of how this can be used (e.g. by Grafana Sift
here).

Commit:
I18n: Crowdin sync (#78402)

* New translations grafana.json (Spanish)

* New translations grafana.json (Chinese Simplified)

* Github Action: Auto-fix i18n files

* New translations grafana.json (Chinese Simplified)

* Github Action: Auto-fix i18n files

---------

Co-authored-by: grafana-i18n-bot <grafana-i18n-bot@users.noreply.github.com>

torkelo committed:
DashboardScene: Basics stubs for starting with editviews (dashboard settings) (#78209)

Commit:
I18n: Crowdin sync (#78418)

* New translations grafana.json (Chinese Simplified)

* Github Action: Auto-fix i18n files

---------

Co-authored-by: grafana-i18n-bot <grafana-i18n-bot@users.noreply.github.com>

adela-almasan committed:
Auto-generate: Scrollable history container (#77895)

inf0rmer committed:
Remove note that mentions Hashicorp Vault support in Cloud (#78327)

torkelo committed:
Explore: Fixes issue with adhoc filters when coming from dashboards (#78339)

bohandley committed:
Prometheus: Fix UI bug where a label with empty string shows as populated with the deleted label filter value (#78153)

* preserve label with empty string

* fix side effect of having undefined show up in the editor select for value

* re render label item when label filters change

yangkb09 committed:
Chore: Deprecate ID from Folder (#78281)

* Chore: Deprecate ID from Folder

* chore: add more linter comments

* chore: add missing lint comment

darrenjaneczek committed:
fix: `label_values` w/ empty filters in prom datasource variable query (#78242)

* fix: handle empty filters in prometheus datasource variable query function `label_values`

kevinwcyu committed:
CloudWatch: Call query method from DataSourceWithBackend to support public dashboards (#77532)

* CloudWatch: call query method from DataSourceWithBackend to support public dashboards

* add test

* remove unneeded properties from test case

* update betterer

* add parens to group related logic

* remove unnecessary aliasing of variable

* use t.Cleanup

* remove redundant check

* add comment

dmihai committed:
Add more unit tests for removeSSOSettings method (#78329)

* add more unit tests for remove sso settings

* add unit tests for the service method

ivanahuckova committed:
Loki: Use TemplateSrv from @grafana/runtime (#78196)

Loki: Use TemplateSrv from grafana/runtime

svennergr committed:
Loki: Fix scope of `labels` variable being wrong (#78417)

fix scope of labels var

Commit:
Changelog: Updated changelog for 10.2.2 (#78439)

Co-authored-by: grafanabot <bot@grafana.com>

bobrik committed:
Dashboards: Implement natural sort for query variables (#78024)

Commit:
I18n: Crowdin sync (#78427)

* New translations grafana.json (German)

* Github Action: Auto-fix i18n files

* New translations grafana.json (German)

* Github Action: Auto-fix i18n files

* New translations grafana.json (Chinese Simplified)

* Github Action: Auto-fix i18n files

* New translations grafana.json (Chinese Simplified)

* Github Action: Auto-fix i18n files

* New translations grafana.json (Chinese Simplified)

* Github Action: Auto-fix i18n files

---------

Co-authored-by: grafana-i18n-bot <grafana-i18n-bot@users.noreply.github.com>

javiruiz01 committed:
[sceneVariablesSetToVariables]: Manually check variable type, instead of instanceof (#78407)

Check type instead of instanceof, manually type variable

ifrost committed:
Chore: update latest.json to 10.2.2 (#78445)

kalleep committed:
Login: remove unused function (#78442)

* Move test to the db so we test the queries and not just testing the mock

* Remove unused function and dependencies

* Remove unused functions from the database

* Add some integration tests

andresmgot committed:
Bug: Fix loading behavior with FlagExternalCorePlugins (#78388)

joshhunt committed:
I18n: Set description of Crowdin PR using GH Actions (#78424)

* I18n: Set description of Crowdin PR using GH Actions

* pin action sha

* Apply suggestions from code review

Co-authored-by: Alex Khomenko <Clarity-89@users.noreply.github.com>

---------

Co-authored-by: Alex Khomenko <Clarity-89@users.noreply.github.com>

andresmgot committed:
Chore: Change endpoint to check versions in cli (#78008)

aditipatelpro committed:
fix: Explore -Tables missing vertical spacing (#78149)

* fix: added vertical space to tables

* chore: used theme for spacing

* fix: extended themeable2

* test: fixed TableContainer test with theme

brendamuir committed:
Docs: adds alerting provisioning updates (#78454)

* Docs: adds alerting provisioning updates

* prettier and changed passives

eleijonmarck committed:
WIP: showing anon users in stats

WIP

create the anon table

matyax committed:
Loki: Fix conditional for derived fields using regex type (#78464)

Derived fields: fix conditional for regex fields

kalleep committed:
Login: refactor auth info package (#78459)

* Remove unused stats and metrics

* No longer collect metrics

* Remove unused dependency

* Move database from sub package

Commit:
I18n: Crowdin sync (#78455)

* New translations grafana.json (French)

* Github Action: Auto-fix i18n files

---------

Co-authored-by: grafana-i18n-bot <grafana-i18n-bot@users.noreply.github.com>

kalleep commented on pkg/api/api.go:
```diff
@@ -578,6 +580,7 @@ func (hs *HTTPServer) registerRoutes() {
 		adminRoute.Get("/settings", authorize(ac.EvalPermission(ac.ActionSettingsRead)), routing.Wrap(hs.AdminGetSettings))
 		adminRoute.Get("/settings-verbose", authorize(ac.EvalPermission(ac.ActionSettingsRead)), routing.Wrap(hs.AdminGetVerboseSettings))
 		adminRoute.Get("/stats", authorize(ac.EvalPermission(ac.ActionServerStatsRead)), routing.Wrap(hs.AdminGetStats))
+		adminRoute.Get("/anonstats", authorize(ac.EvalPermission(ac.ActionServerStatsRead)), routing.Wrap(hs.GetAnonUsers))
```
We should put api routes in the anonymous package as we do for our other services, and move the handlers there as well

gelicia committed:
Explore: Default synced to true, only show synced status if panes are split (#77759)

* Default synced to true, only show synced status if panes are split

* Don’t default sync to true until split happens, handle syncing when init or history change sync happens

* Sync times if origin pane and new pane have the same range

kaydelaney committed:
Chore: Clean up usage of deprecated stylesFactory function (#78419)

gamab committed:
RBAC: Check `plugins:install` globally (#78438)

* RBAC: Check plugins:install globally

* Add disclamer to the RBACSingleOrganization config option

torkelo committed:
DashboardScene: Fixes wrapping panel time override (#78470)

Commit:
I18n: Crowdin sync (#78471)

* New translations grafana.json (Spanish)

* New translations grafana.json (German)

* Github Action: Auto-fix i18n files

---------

Co-authored-by: grafana-i18n-bot <grafana-i18n-bot@users.noreply.github.com>

ashharrison90 committed:
Loading screen: Ensure bounce animation isn't blocked by main thread (#78469)

remove transform-origin to ensure animation gets put on compositor thread

juanicabanas committed:
Revert "Panel: omit query API call when the panel is a row (#75847)" (#78475)

torkelo committed:
DashboardScene: Fixes focus panel z-index issue (#78474)

ivanahuckova committed:
Loki: Change import for CustomVariableModel and remove unnecessary import of initialCustomVariableModelState (#78456)

leeoniya committed:
HeatmapTooltip: Refactor to simplify upcoming PRs (#78436)

harisrozajac committed:
Explore: Wrap RawPrometheusContainer with PanelChrome (#78084)

* Use LoadingState type for loading prop

* Fix test

julienduchesne committed:
Swagger: Rename API Key `AddCommand` (#78491)

As a global definition, `AddAPIKeyCommand` is clearer

baldm0mma committed:
Chore: Update e2e selectors for Transformations (#78426)

* baldm0mma/e2e_selector_migration/ update labels

* baldm0mma/e2e_selector_migration/ run prettier

* baldm0mma/e2e_selector_migration/ add test

* baldm0mma/ update results

* baldm0mma/e2e_selector_migration/ update betterer

mgyongyosi committed:
Auth: Use camelCase in SSO Settings API (#78480)

Return/accept camelCase fields in sso setting api

lovasoa committed:
PublicDashboards: Typo in acknowledge checkboxes information (#77375)

Co-authored-by: Isabel Matwawana <isabel.matwawana@grafana.com>
Co-authored-by: Juan Cabanas <juan.cabanas@grafana.com>

baldm0mma committed:
Baldm0mma/docs expansion (#78142)

* baldm0mma/docs_expansion/ update Convert field type

* baldm0mma/docs_expansion/ update Extract fields

* baldm0mma/docs_expansion/ update Lookup fields from resource

* baldm0mma/docs_expansion/ update Filter data by query refId

* baldm0mma/docs_extension/ update Filter fields by name

* baldm0mma/docs_expansion/ update Format time

* baldm0mma/docs_expansion/ update Grouping to matrix

* baldm0mma/docs_expansion/ update Create heatmap

* baldm0mma/docs_extension/ updte Histogram

* baldm0mma/docs_expansion/ update Join by field

* baldm0mma/docs_expansion/ update Join by labels

* baldm0mma/docs_expansion/ update Labels to fields

* baldm0mma/docs_expansion/ update Limit

* baldm0mma/docs_expansion/ update Merge

* baldm0mma/docs_expansion/ update Oraganize fields

* baldm0mma/docs_expansion/ update Partition by values

* baldm0mma/docs_expansion/ update Prepare time series

* baldm0mma/docs_expansion/ update Rename by regex

* baldm0mma/docs_expansion/ update Rows to fields

* baldm0mma/docs_expansion/ update Series to rows

* baldm0mma/docs_expansion/ update Time series to table transform

* baldm0mma/docs_expansion/ build docs

* baldm0mma/docs_expansion/ update Prepare time series

* baldm0mma/docs_expansion/ build markdown

* Update public/app/features/transformers/docs/content.ts

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* Update public/app/features/transformers/docs/content.ts

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* Update public/app/features/transformers/docs/content.ts

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* Update public/app/features/transformers/docs/content.ts

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* Update public/app/features/transformers/docs/content.ts

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* Update public/app/features/transformers/docs/content.ts

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* baldm0mma/docs_expansion/ update Rows to fields conclusion

* baldm0mma/docs_expansion/ update sort by

* Update public/app/features/transformers/docs/content.ts

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* Update public/app/features/transformers/docs/content.ts

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* baldm0mma/docs_expansion/ build markdown

* baldm0mma/docs_expansion/ update note in Add field from calculation

* baldm0mma/docs_expansion/ update spelling

* baldm0mma/docs_expansion/ update wording

* Update public/app/features/transformers/docs/content.ts

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* Update public/app/features/transformers/docs/content.ts

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* Update public/app/features/transformers/docs/content.ts

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* Update public/app/features/transformers/docs/content.ts

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* Update public/app/features/transformers/docs/content.ts

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* Update public/app/features/transformers/docs/content.ts

* Update public/app/features/transformers/docs/content.ts

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* Update public/app/features/transformers/docs/content.ts

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* Update public/app/features/transformers/docs/content.ts

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* Update public/app/features/transformers/docs/content.ts

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* Update public/app/features/transformers/docs/content.ts

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* Update public/app/features/transformers/docs/content.ts

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* Update public/app/features/transformers/docs/content.ts

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* Update public/app/features/transformers/docs/content.ts

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* Update public/app/features/transformers/docs/content.ts

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* baldm0mma/docs_expansion/ update Partition by values

* baldm0mma/docs_expansion/ update enhancements

* Update public/app/features/transformers/docs/content.ts

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* baldm0mma/docs_expansion/ rem whitesapce

* baldm0mma/docs_expansion/ build markdown

* Update public/app/features/transformers/docs/content.ts

* baldm0mma/docs_expansion/ update whitespace

* baldm0mma/docs_expansion/ update Extract fields

* baldm0mma/docs_expansion/ update with @codeincarnate's suggestions

* Update public/app/features/transformers/docs/content.ts

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* Apply suggestions from code review

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

* baldm0mma/docs_expansion/ update syntax

* baldm0mma/docs_expansion/ update transformation names

* baldm0mma/docs_expansion/ build markdown

* baldm0mma/docs_expansion/ dataset fix

* baldm0mma/docs_expansion/ build docs

---------

Co-authored-by: Isabel <76437239+imatwawana@users.noreply.github.com>

juanicabanas committed:
PublicDashboards: Middleware creation (#77941)

undef1nd committed:
Nested Folders: Wrap create/update operations with transactions (#78000)

* Nested Folders: Add transaction to create and update methods

* Update tests

* Make IncreaseVersionForAllRulesInNamespace synchronous

* Resolve merge conflicts

Commit:
I18n: Crowdin sync (#78485)

* New translations grafana.json (Spanish)

* Github Action: Auto-fix i18n files

* New translations grafana.json (French)

* New translations grafana.json (Spanish)

* Github Action: Auto-fix i18n files

* New translations grafana.json (French)

* New translations grafana.json (Spanish)

* Github Action: Auto-fix i18n files

* New translations grafana.json (French)

* New translations grafana.json (Spanish)

* Github Action: Auto-fix i18n files

* New translations grafana.json (French)

* New translations grafana.json (Spanish)

* Github Action: Auto-fix i18n files

* New translations grafana.json (Spanish)

* Github Action: Auto-fix i18n files

---------

Co-authored-by: grafana-i18n-bot <grafana-i18n-bot@users.noreply.github.com>

andresmgot committed:
Chore: Display core plugins when showing all in the catalog (#78447)

torkelo committed:
DashboardScene: Fixes issue with variables without current property (#78481)

torkelo committed:
DashboardScene: Link to explore should take adhoc filters into account (#78440)

* DashboardScene: Link to explore should take adhoc filters into account

* Fix imports

javiruiz01 committed:
[sceneVariablesSetToVariables]: Use type guards from scenes to type variable (#78488)

* Update grafana/scenes version to 1.24.0

* Use type guards to properly type variables

* Update scenes to version 1.24.1

Develer committed:
Status History: Improve tooltip (#76647)

* feat(state-timeline-tooltip): improve status history tooltip

Co-authored-by: Adela Almasan <adela.almasan@grafana.com>
Co-authored-by: Leon Sorokin <leeoniya@gmail.com>

joey-grafana committed:
Tempo: Switch to sdk httpclient from infra httpclient (#78446)

Switch to sdk httpclient from infra httpclient

volcanonoodle committed:
AuthN: Support HA setups with External Service Account management (#78425)

* Lock when creating external service

* Add local lock back

* Improve function signature

* Define lockName separately to make it more explicit

* Update pkg/infra/serverlock/serverlock.go

Co-authored-by: Gabriel MABILLE <gamab@users.noreply.github.com>

* Update pkg/infra/serverlock/serverlock.go

Co-authored-by: Gabriel MABILLE <gamab@users.noreply.github.com>

---------

Co-authored-by: Gabriel MABILLE <gamab@users.noreply.github.com>

soniaAguilarPeiron committed:
Alerting: Simplify routing in alert form - part1 (#78040)

* Add routing option tabs

* Use alertingSimplifiedRouting feature toggle

* Move simplified routing tab to a separate component:SimplifiedRouting

* Populate contact point selector with the right values

* Show alert manager icons

* Fix descriptions

* Remove clear button on ContactPointSelector and save updated reducer state in the form

* Load contact points and manual option from rule data in RuleFormValues

* make contact point selector not clearable

* Refactor

* Add link to contact points view

* Move ContactPointSelector to a separate file

* Refactor: move hoook useReceiversMetadataMapByName to a separate file

* Update Need more info texts

* Address some PR review comments

* Use useContactPointsWithStatus hook and wrap each ContacPointSelector with AlertmanagerProvider

* use getAlertManagerDataSourcesByPermission instead of useGetAlertManagersMetadata in NotificationPreview

* Update enum

* Remove css style

* remove console

* update contact point selector

* file cleanup

* adds summary as description

* Update text in manual tab

* Fix preview routing not checking if alert manager can handle grafana alerts

* Fix typo

* remove unused location form field

* fix prettier

* fix test

* Remove unused location form field from AlertRuleNameInput

* Only use internal AlertManager for now

---------

Co-authored-by: Gilles De Mey <gilles.de.mey@gmail.com>

IevaVasiljeva committed:
Data sources: Refactor logic for naming new data sources (#78479)

* move the name finding logic for new data sources from frontend to backend

* cleanup and fix test

* linting

* change the way the number after the ds type is incremented - keep incrementing it without adding more hyphens

* enterprise spec updates (unrelated to the PR)

ivanahuckova committed:
`@grafana/data`: Add `serializeParams` (#78468)

* Move serializeParams to @grafana/data

* Update comment

* Update solution in Tempo

* Fix type assertions

* Use toUrlParams in serializeParams

* Update data sources

* Update

* Update packages/grafana-data/src/utils/url.ts

leventebalogh committed:
Plugins: Support markdown in custom plugin deprecation messages (#78251)

* feat: support markdown in custom deprecation messages

* fix: use a div wrapper for rendered markdown

* feat: update the default deprecation message

dprokop committed:
DashboardScene: Support Textbox variables (#78525)

gtk-grafana committed:
Logs Panel: Table UI - Time range changes not reflecting in table (#78500)

* add hook to update selected dataframe when time range changes

Jguer committed:
Authz: Remove use of SignedInUser copy for permission evaluation (#78448)

* remove use of SignedInUserCopies

* add extra safety to not cross assign permissions

unwind circular dependency

dashboardacl->dashboardaccess

fix missing import

* correctly set teams for permissions

* fix missing inits

* nit: check err

* exit early for api keys

ivanahuckova committed:
Loki: Remove relying on timeSrv.timeRange in LanguageProvider (#78450)

* Loki: Allow setting of timeRange when using languageProvider functions

* Loki: Use timerange where available for start

* Loki: Use timerange where available for fetchLabels

* Loki: Use timerange where available for fetchSeriesLabels

* Loki: Use timerange where available for fetchLabelValues

* Loki: Use timerange where available for getParserAndLabelKeys

* Loki: Update and add tests for fetchLabels

* Loki: Update and add tests for fetchSeriesLabels

* Loki: Update and add tests for fetchSeries

* Loki: Update and add tests for fetchLabelValues

* Loki: Update and add tests for fetchLabelValues

* Loki: Update and add tests for getParserAndLabelKeys

* Update public/app/plugins/datasource/loki/LanguageProvider.test.ts

Co-authored-by: Matias Chomicki <matyax@gmail.com>

* Update public/app/plugins/datasource/loki/LanguageProvider.test.ts

Co-authored-by: Matias Chomicki <matyax@gmail.com>

* Not needing to use languageProvider.getDefaultTime in Monaco

* Update comment

* Update getDefaultTimeRange to be ptivate

---------

Co-authored-by: Matias Chomicki <matyax@gmail.com>

dmihai committed:
Add unit tests for the removeSSOSettings API method (#78476)

add unit tests for the removeSSOSettings api method

itsmylife committed:
Chore: Don't show hint for recorded queries (#78422)

Don't show hint on recorded queries

Jguer committed:
Live: Move empty orgRole safety valve (#78531)

move empty orgRole safety valve

ashharrison90 committed:
Chore: Remove `newBrowseDashboards` feature toggle (#78190)

* remove all the things

* fix OldFolderPicker tests

* i18n

* remove more unused code

* remove mutation of error object since it's now frozen in the redux state

* fix error handling

ashharrison90 committed:
Breadcrumbs: Only dedupe the lowest child node of `sectionNav` (#78279)

only dedupe sectionNav leaf node

ashharrison90 committed:
DockedMegaMenu: Keep undock button (#78461)

* dock undock smoothly

* handle keyboard focus

* use ref instead of state

* run i18n:extract

* undo this change

* make dock/undock first button to focus

* only focus when going to docked, add comment

* minor tweaks

oscarkilhed committed:
Avatar: Allow browser caching of /avatar/ (#78314)

don't set no-store headers to /avatar/

petethepig committed:
FlameGraph: reduce max height of table (#78432)

* FlameGraph: reduce max height of table

* prettier fix

* use maxHeight

* makes it actually work

* makes things actually work

---------

Co-authored-by: Leon Sorokin <leeoniya@gmail.com>

santihernandezc committed:
Alerting: Fix remote Alertmanager integration tests (CI/Makefile) (#78541)

* Alerting: Fix remote Alertmanager integration tests (CI/Makefile)

itsmylife committed:
InfluxDB: Parse data for table view to have parity with frontend parser (#78365)

* Use TimeSeriesWide format for table response

* fix group by query result parsing

* handle labels

* provide a test where result has no tags

* parsing results without time column

* clean the code

* remove the comment line

* more cleaning

* lint

thepalbi committed:
CloudWatch: Add AWS Bedrock metrics definition (#78478)

imatwawana committed:
docs: 404 fixes - what's new Alerting content (#78549)

* Updated links in what's new 10.2 and what's new next

* Attempted to add alias

* Fix alias

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

---------

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>
Co-authored-by: Jack Baldry <jack.baldry@grafana.com>

nmarrs committed:
XYChart: Minor improvements (#78506)

Co-authored-by: Leon Sorokin <leeoniya@gmail.com>
Co-authored-by: Adela Almasan <adela.almasan@grafana.com>

toddtreece committed:
K8s: Prevent user impersonation (#78555)

adela-almasan committed:
Timeseries: Remove cursor sync when x is not time (#78496)

itsmylife committed:
InfluxDB SQL: Fix sql query template variable interpolation (#78557)

interpolate sql too

thiskevinwang committed:
chore(authn.service): fix typo in log statement (#76205)

sunker committed:
Variables: Remove alpha flag from variable support API (#78573)

remove alpha flag

Jguer committed:
SignedInUser: Only register echo backend if user is signed in (#78545)

fix SignedIn checks

ivanahuckova committed:
Loki: Move `convertToWebSocketUrl` from Explore to Loki (#78542)

* Move convertToWebSocketUrl to Loki

* Add tests

* Fix test

matyax committed:
Logs: create DataSourceWithQueryModificationSupport + determine popover menu support (#78322)

* DataSourceWithQueryModificationSupportSupport: create interface

* Loki: implement DataSourceWithQueryModificationSupportSupport

* Elasticsearch: implement DataSourceWithQueryModificationSupportSupport

* DataSourceWithQueryModificationSupportSupport: add type guard

* DataSourceWithQueryModificationSupport: rename

* Check for nullish values in guards

* Logs container: replace support map with ds instances map

* Log rows: refactor deselection listener

* Formatting

* Formatting

* DataSourceWithQueryModificationSupport: add missing comment

* Logs container: update method name

* Logs container: check for query modification support

* Create QueryFixType

* QueryFixAction: move back to ds types

* getSupportedQueryModifications: update signature

* getSupportedQueryModifications: update signature

joshhunt committed:
Chore: Refactor TimeRangePicker for aria-label selectors  (#78399)

* Change from aria-label to data-testid for e2e selectors

* translate

* update tests

* swap buttons:

IevaVasiljeva committed:
Annotations: Remove dashboard permission checks for annotations (#78352)

remove checks for access to dashboard if FlagAnnotationPermissionUpdate is enabled

ashharrison90 committed:
Tab: Only make `Tab` an anchor if a `href` is passed (#78540)

* initial work

* only make the tab an anchor if it has a href

* move things around for smaller diff

* use content()

* eslint-disable the type assertions

* extract props into common object and add missing return statement

gamab committed:
RBAC: GrafanaAdmin users are admins of the Global Organization (#78559)

dprokop committed:
DashboardScene: Share library panel (#78421)

* DashboardScene: Share library panel

* Add menu item to create library panel

* Test update

bohandley committed:
Opentsdb: Add variables to select when interacting with the metric select (#78558)

add variables to select when interacting with the metric select

domasx2 committed:
Chore: Update "Observability" nav group subtitle (#78579)

Update observability nav group subtitle

svennergr committed:
Loki: Add structured metadata keys to autocomplete (#78584)

* wip

* remove import

* scope to monaco completions for now

* use `LabelType` enum

* change strucutred metadata documentation

* fix import

* add `responseUtils` tests

* update tests

* fix completions.ts tests

* fix LabelType enum

* fix CompletionDataProvider test

joey-grafana committed:
Tempo: Embed flame graph in span details (#77537)

* Embed flame graph

* Update test

* Update test

* Use toggle

* Update test

* Add tests

* Use const

* Cleanup

* Update profile tag

* Move flame graph out of tags, remove request and other cleanup + tests

* Update test

* Set flame graph by profile id and simplify logic

* Cleanup and redrawListView

* Create/use feature toggle

svennergr committed:
Loki: improve structured metadata in devenv (#78587)

AgnesToulet committed:
Doc: Update image rendering configuration (#77042)

* Doc: Update image rendering configuration

* Update docs/sources/setup-grafana/configure-grafana/_index.md

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

* Update docs/sources/setup-grafana/configure-grafana/_index.md

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

* Update docs/sources/setup-grafana/image-rendering/_index.md

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

---------

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

torkelo committed:
DashboardScene: Fixes angular panels inside collapsed rows (#78532)

* DashboardScene: Fixes angular panels inside collapsed rows

* Update

ivanahuckova committed:
Loki: Fix trailing comma in renderers for `json`, `logfmt`, `keep` and `drop` operations (#78594)

* Loki: Fix renderers for json, logfmt, keep and drop pipe operations

* Make spacing and trimming more consistent

* Fix

mgyongyosi committed:
Auth: Add more context to logs around token rotation, revocation (#78600)

Add more context to logs around token rotation, revocation

Co-authored-by: Karl Persson <kalle.persson@grafana.com>

jackw committed:
Chore: Refresh lock file with correct scenes version info (#78605)

chore(yarn): refresh lock file with correct scenes version info

gotjosh committed:
Alerting: Introduce a Mimir client as part of the Remote Alertmanager (#78357)

* Alerting: Introduce a Mimir client as part of the Remote Alertmanager

This is our first attempt at making Grafana communicate use Mimir as a backend - it uses a new set of APIs that we've developed on the Mimir side to upload the grafana configuration and alertmanager state so that it can then be ported over.

Codewise, we've introduced a couple of things:

A client to isolate in its own package all the communication that happens with Mimir
A few changes to the remote/alertmanager to include uploading the configuration and state when it starts
A few refactors that align a bit better with the design approach that we're thinking
An integration tests again these newly developed APIs using a custom image

---------

Signed-off-by: gotjosh <josue.abreu@gmail.com>
Co-authored-by: Santiago <santiagohernandez.1997@gmail.com>

JohnnyQQQQ committed:
fix(alerting): proper handling for queries with multiple conditions in migration (#78591)

fix(alerting): proper handling for queries with multiple conditions

Jguer committed:
AuthProxy: Do not allow sessions to be assigned with other methods (#78602)

do not allow login token with other methods

dmihai committed:
Auth: Add more tests for the SSO settings upsert function (#78544)

* add more tests for the SSO settings upsert function

* fix client id to match provider

* use time now in tests

gabor committed:
postgres: update unit test to be ready for the xorm-removal (#78537)

postgres: removed xorm from unit tests

eleijonmarck committed:
Add anonymous stats and user table

- anonymous users users page
- add feature toggle `anonymousAccess`
- remove check for enterprise for `Device-Id` header in request
- add anonusers/device count to stats
